### DISCUSSION TITLE
Extending Gov4Git to be more agnostic

### DISCUSSION THREAD

**Discussion by BastinJafari** [gov4git/gov4git#176] [Ideas]
Hello folks,

I recently stumbled over your paper and Gov4Git and was incredibly inspired by them. It seems to be the missing solution to a problem for a project of mine in a similar context. 

The project I envision is a central knowledge base for all communities that try to solve the [meta-crisis](https://www.sloww.co/meta-crisis-101/). In my research around that idea space, I encountered countless actors, communities, and institutions that all have in common to work on solving that problem from different angles. Often, these communities are unaware of most of the other players or that they are even part of a common idea space.
So, the development of problem definitions, concepts, and solutions is more atomized than it should be.

The problem I faced was which decentralized system to use to manage write access and management to that knowledge base. That means determining which community/institution/actors are part of the common idea space and, therefore, should be invited into the knowledge base and how to decide who from this set of communities should manage the infrastructure and administration.

That is now solvable with the plural management protocol.

Although, there seems to be only an implementation for Open-Source Project Governance for now, which doesn't work perfectly with the concept of a knowledge base for heterogeneous communities. There are a few assumptions that cannot be made. Unlike open-source projects, communities do not necessarily agree that the identity, information, and infrastructure layer should be Git and GitHub. It could be all kinds of systems, like a WordPress Website with MongoDB and [SoulBound Tokens](https://medium.com/coinmonks/souls-soulbound-tokens-f65b5b4f1c2a) for identity management.

It's unclear how to modify Gov4Git to suit that.

So I would like to start a discussion about that problem.
Are there thoughts on building a new, more general layer-agnostic framework? 
If yes, is it reasonable to keep Git as the base information layer and then map other layers (Identity, Databases, Blockchain, IPFS, etc.) to it, or should it be further abstracted?

Please let me know your thoughts!

**Comment by petar** [comment]
Hi @BastinJafari, thank you for your note.

First, I wanted to mention that as of v2.2.0, both the Plural Management Protocol and the Waimea Protocol (for governing open source communities) are supported by Gov4Git. The PMP corresponds to the label `gov4git:pmp-v1`, and Waimea is the label `gov4git:waimea`.

git is the storage backend that Gov4Git needs to store its internal records. This does not preclude the actual collaboration (the user-facing interface) to happen on any platform where this makes sense. A discussion on this subject would be interesting, especially envisioning the UX flow.

To answer your other question: Gov4Git is compatible with other systems, in the sense that you could build a bridge between Gov4Git and a blockchain, for example. Furthermore, any ZK-based primitives (such as soulbound tokens) can be implemented and used inside Gov4Git, as they don't rely on the nature of the storage substrate.

While the Gov4Git Foundation does not have the resources to undertake this kind of project at the moment, we welcome such initiatives from outside collaborators.
